crates/repo/src/lib.rs

```diff
@@ -279,4 +220,40 @@ mod tests {
             })
         );
     }
+
+    #[tokio::test]
+    async fn routes() {
+        let listener = TcpListener::bind("127.0.0.1:8000".parse::<SocketAddr>().unwrap()).unwrap();
+        let addr = listener.local_addr().unwrap();
+
+        tokio::spawn(async move {
+            axum::Server::from_tcp(listener)
+                .unwrap()
+                .serve(app().into_make_service())
+                .await
+                .unwrap();
+        });
+
+        let client = hyper::Client::new();
+
+        let response = client
+            .request(
+                Request::builder()
+                    .method("PUT")
+                    .header(
+                        header::CONTENT_DIGEST,
+                        ":mqVuAfXRKap7bdgcCY5uykM6+R9GqQ8K/uxy9rx7HNQlGYl1kPzQho1wx4JwY8w=:",
+                    )
+                    .header(header::CONTENT_TYPE, "text/plain")
+                    .header(header::CONTENT_LENGTH, "16")
+                    .uri(format!("http://{}/test/onetwothree", addr))
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        // TODO: Validate output for successful namespace creation
+        let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
```

Comment:
What I would expect to happen here, would be a `GET` on non-existing repo, asserting a 404, then `PUT` to that repo asserting 200, then `GET` on the same repo now succeeding and asserting 200.